tests/api/test_flexible_calling.py

```diff
@@ -0,0 +1,160 @@
+            json_case_path = "./tests/api/data/flexible_calling_unit_test/verification_case_unit_test_Path.json"
+
+            # Change working_dir value in the json file to a valid path in Linux format
+            with open(json_case_path, "r") as f:
+                workflow_dict = json.load(f)
+                workflow_dict["working_dir"] = "./tests/api/result"
+
+            with open(json_case_path, "w") as f:
+                json.dump(workflow_dict, f)
+
+            workflow = Workflow(workflow=json_case_path)
+            # change current working directory back
+            os.chdir("../../..")
+
+            self.assertEqual(
+                logobs.output[0],
+                "INFO:root:the working dir provided is in Linux format.",
+            )
+
+    def test_Win_path(self):
+        """This test check if the program can detect the working path provided is in WIN format."""
+        with self.assertLogs() as logobs:
+            json_case_path = "./tests/api/data/flexible_calling_unit_test/verification_case_unit_test_Path.json"
+
+            # Change working_dir value in the json file to a valid path in Win format
+            with open(json_case_path, "r") as f:
+                workflow_dict = json.load(f)
+                workflow_dict["working_dir"] = ".\\tests\\api\\result"
+
+            with open(json_case_path, "w") as f:
+                json.dump(workflow_dict, f)
+
+            workflow = Workflow(workflow=json_case_path)
+            # change current working directory back
+            os.chdir("../../..")
+
+            self.assertEqual(
+                logobs.output[0],
+                "INFO:root:the working dir provided is in Win format.",
+            )
+
+    def test_dir_not_exist(self):
+        """This test checks when a valid wd is provided but it doesn't exist,
+        if the program will behave correctly"""
+        with self.assertLogs() as logobs:
+            json_case_path = "./tests/api/data/flexible_calling_unit_test/verification_case_unit_test_Path.json"
+
+            # Change working_dir value in the json file to a path that does not exist
+            with open(json_case_path, "r") as f:
+                workflow_dict = json.load(f)
+                workflow_dict["working_dir"] = "./not_existing_path/test"
+
+            with open(json_case_path, "w") as f:
+                json.dump(workflow_dict, f)
+
+            workflow = Workflow(workflow=json_case_path)
+            self.assertEqual(
+                logobs.output[1],
+                "ERROR:root:working directory specified does not exist.",
+            )
+
+    def test_valid_dir(self):
+        """This test checks when a working directory is provided and it also points to the correct path,
+        if the program will behave correctly"""
+        with self.assertLogs() as logobs:
+            json_case_path = "./tests/api/data/flexible_calling_unit_test/verification_case_unit_test_Path.json"
```

Comment:
seems like all tests are about relative path. Please also make sure it works with absolute path